src/rdb.c

```diff
@@ -3952,3 +3952,41 @@ rdbSaveInfo *rdbPopulateSaveInfo(rdbSaveInfo *rsi) {
     }
     return NULL;
 }
+
+/* Restore the replication ID / offset from the RDB file
+ * return 1 if rdbSaveInfo is valid */
+int replicationRestoreOffsetFromSaveInfo(rdbSaveInfo *rsi, bool is_aof_preamble) {
+    int rsi_is_valid = 0;
+    if (rsi == NULL) return rsi_is_valid;
```

Comment:
We always pass a valid `rsi` pointer, don't we? No need to allow passing NULL. We can do an assert instead.

```suggestion
    serverAssert(rsi != NULL);
```

Reply:
This is a defensive code, but I think `assert` makes sense